Utils.py

```diff
@@ -513,10 +513,13 @@ def filter(self, record: logging.LogRecord) -> bool:
             return self.condition(record)
 
     file_handler.addFilter(Filter("NoStream", lambda record: not getattr(record,  "NoFile", False)))
+    file_handler.addFilter(Filter("NoCarriageReturn", lambda record: '\r' not in record.msg))
```

Comment:
What's this about? Hard to figure out from reading the docs why this is here

Reply:
'\r' (or carriage return) returns the cursor to the left of the line without line feeding ('\n'). For console based output this is fine, as it just overwrites the line in place. When logging to a file however it messes up the logs frequently rendering them illegible or useless.

This was an accidental commit from another project, but it seems like a prudent addition as it attempts to protect logs from people potentially making mistakes. I'll take it out if requested.